### PR TITLE
8321059: Unneeded array assignments in MergeCollation and CompactByteArray

### PR DESCRIPTION
**Pull request by naotoj**
Removing the unnecessary array assignments.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8321059](https://bugs.openjdk.org/browse/JDK-8321059): Unneeded array assignments in MergeCollation and  CompactByteArray (**Enhancement** - P4)

### Reviewers
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer) ⚠️ Review applies to [4fd6e61b](https://git.openjdk.org/jdk/pull/16912/files/4fd6e61be63d443787607fcd0d488efa3982de2a)
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16912/head:pull/16912` \
`$ git checkout pull/16912`

Update a local copy of the PR: \
`$ git checkout pull/16912` \
`$ git pull https://git.openjdk.org/jdk.git pull/16912/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16912`

View PR using the GUI difftool: \
`$ git pr show -t 16912`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16912.diff">https://git.openjdk.org/jdk/pull/16912.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16912#issuecomment-1834543282)

### CODE REVIEW

**Review by bplb — APPROVED** [review]
Looks fine.

Is that "XXX" reminder at line 58 in MergeCollation still pertinent or could it be removed?

File: src/java.base/share/classes/sun/text/CompactByteArray.java [hunk 1]
```diff
@@ -77,12 +77,13 @@ public CompactByteArray(byte defaultValue)
         values = new byte[UNICODECOUNT];
         indices = new short[INDEXCOUNT];
         hashes = new int[INDEXCOUNT];
-        for (i = 0; i < UNICODECOUNT; ++i) {
-            values[i] = defaultValue;
+        if (defaultValue != (byte)0) {
+            for (i = 0; i < UNICODECOUNT; ++i) {
+                values[i] = defaultValue;
+            }
```
Comment (bokken) [inline src/java.base/share/classes/sun/text/CompactByteArray.java:83]: should this be `Arrays.fill(values, defaultValue);` ?

https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/Arrays.html#fill(byte%5B%5D,byte)

**Comment by naotoj** [comment]
> Is that "XXX" reminder at line 58 in MergeCollation still pertinent or could it be removed?

Maybe, but not 100% sure, so I just left it.

**Review by bplb — APPROVED** [review]

**Comment by bplb** [comment]
> > Is that "XXX" reminder at line 58 in MergeCollation still pertinent or could it be removed?
> 
> Maybe, but not 100% sure, so I just left it.

If unsure, better to leave it.

**Review by RogerRiggs — APPROVED** [review]
LGTM

**Pull request closed without merging**